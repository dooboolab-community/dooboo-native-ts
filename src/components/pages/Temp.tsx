import {Button, Icon, useTheme} from 'dooboo-ui';
import type {
  RootStackNavigationProps,
  RootStackParamList,
} from '../navigations/RootStack';

import React from 'react';
import type {RouteProp} from '@react-navigation/core';
import styled from '@emotion/native';

const Container = styled.View`
  flex: 1;
  align-self: stretch;

  background-color: ${({theme}) => theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

interface Props {
  navigation: RootStackNavigationProps<'Temp'>;
  route: RouteProp<RootStackParamList, 'Temp'>;
}

function Page(props: Props): React.ReactElement {
  const {theme} = useTheme();

  const {
    route: {
      params: {param},
    },
    navigation,
  } = props;

  return (
    <Container>
      <Button
        testID="btn-back"
        onPress={(): void => navigation.goBack()}
        text={param}
        startElement={
          <Icon
            name="chevron-left-light"
            color={theme.textContrast}
            style={{
              marginRight: 4,
            }}
          />
        }
      />
    </Container>
  );
}

export default Page;
