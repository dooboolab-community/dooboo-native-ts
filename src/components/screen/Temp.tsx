import { MainStackParamList, RouteProp } from '@react-navigation/core';

import Button from '../shared/Button';
import React from 'react';
import { RootStackNavigationProps } from '../../components/navigation/RootStackNavigator';
import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  background-color: ${(props): string => props.theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

interface Props {
  navigation: RootStackNavigationProps<'Temp'>;
  route: RouteProp<MainStackParamList, 'Temp'>;
}

function Page(props: Props): React.ReactElement {
  const { route: { params: { param } }, navigation } = props;
  return (
    <Container>
      <Button
        testID="btn-back"
        onClick={(): void => navigation.goBack()}
        text={param}
        style={{
          backgroundColor: '#333333',
        }}
      />
    </Container>
  );
}

export default Page;
