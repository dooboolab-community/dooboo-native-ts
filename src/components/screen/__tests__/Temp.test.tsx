import 'react-native';

import React, { ReactElement } from 'react';
import {
  RenderResult,
  act,
  fireEvent,
  render,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Temp from '../Temp';
import { ThemeType } from '@dooboo-ui/native-theme';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: ReactElement;
let testingLib: RenderResult;

describe('[Temp] render', () => {
  props = createTestProps({
    route: {
      params: {
        param: 'GO BACK',
      },
    },
  });
  component = createTestElement(<Temp {...props} />);

  it('renders without crashing', () => {
    testingLib = render(component);
    const { baseElement } = testingLib;
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it('should render [Dark] theme', () => {
    props = createTestProps({
      route: {
        params: {
          param: 'GO BACK',
        },
      },
    });
    component = createTestElement(<Temp {...props} />, ThemeType.DARK);
    testingLib = render(component);
    const { baseElement } = testingLib;
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('[Temp] Interaction', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(component);
  });

  it('should simulate [onClick] when button has been clicked', () => {
    const btnInstance = renderResult.getByTestId('btn-back');
    act(() => {
      fireEvent.press(btnInstance);
    });
    expect(props.navigation.goBack).toHaveBeenCalled();
  });
});
