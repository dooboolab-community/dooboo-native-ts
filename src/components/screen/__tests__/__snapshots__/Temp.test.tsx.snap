// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.5}
        testID="btn-back"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#069ccd",
                "borderColor": "#069ccd",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#333333",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 14,
                },
              ]
            }
          >
            GO BACK
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`[Temp] render should render [Dark] theme 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#323739",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.5}
        testID="btn-back"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#069ccd",
                "borderColor": "#069ccd",
                "borderRadius": 4,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "width": 320,
              },
              Object {
                "backgroundColor": "#333333",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 14,
                },
              ]
            }
          >
            GO BACK
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
